nexus/tests/integration_tests/ip_pools.rs

```diff
@@ -0,0 +1,520 @@
+            description: None,
+        },
+    };
+    let modified_pool: IpPool =
+        NexusRequest::object_put(client, &ip_pool_url, Some(&updates))
+            .authn_as(AuthnMode::PrivilegedUser)
+            .execute()
+            .await
+            .unwrap()
+            .parsed_body()
+            .unwrap();
+    assert_eq!(modified_pool.identity.name, new_pool_name);
+    assert_eq!(modified_pool.identity.id, created_pool.identity.id);
+    assert_eq!(
+        modified_pool.identity.description,
+        created_pool.identity.description
+    );
+    assert_eq!(
+        modified_pool.identity.time_created,
+        created_pool.identity.time_created
+    );
+    assert!(
+        modified_pool.identity.time_modified
+            > created_pool.identity.time_modified
+    );
+
+    let fetched_modified_pool: IpPool =
+        NexusRequest::object_get(client, &new_ip_pool_url)
+            .authn_as(AuthnMode::PrivilegedUser)
+            .execute()
+            .await
+            .unwrap()
+            .parsed_body()
+            .unwrap();
+    assert_pools_eq(&modified_pool, &fetched_modified_pool);
+
+    let error: HttpErrorResponseBody = NexusRequest::expect_failure(
+        client,
+        StatusCode::NOT_FOUND,
+        Method::GET,
+        &ip_pool_url,
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .unwrap()
+    .parsed_body()
+    .unwrap();
+    assert_eq!(
+        error.message,
+        format!("not found: ip-pool with name \"{}\"", pool_name),
+    );
+
+    // Delete the range, then verify we can delete the pool and everything looks
+    // gravy.
+    /*
+     * Most clients should expect 201, which isn't really what we're doing here.
+    NexusRequest::objects_post(client, &new_ip_pool_del_range_url, &range)
+        .authn_as(AuthnMode::PrivilegedUser)
+        .execute()
+        .await
+        .expect("Failed to delete IP range within pool");
+    */
+    NexusRequest::new(
+        RequestBuilder::new(client, Method::POST, &new_ip_pool_del_range_url)
+            .body(Some(&range))
+            .expect_status(Some(StatusCode::NO_CONTENT)),
+    )
+    .authn_as(AuthnMode::PrivilegedUser)
+    .execute()
+    .await
+    .expect("Failed to delete IP range from a pool");
+    NexusRequest::object_delete(client, &new_ip_pool_url)
+        .authn_as(AuthnMode::PrivilegedUser)
+        .execute()
+        .await
+        .expect("Expected to be able to delete an empty IP Pool");
+}
+
+// Data for testing overlapping IP ranges
+struct TestRange {
+    // A starting IP range that should be inserted correctly
+    base_range: IpRange,
+    // Ranges that should fail for various reasons of overlap
+    bad_ranges: Vec<IpRange>,
+}
+
+// Integration test verifying the uniqueness of IP ranges when inserted /
+// deleted across multiple pools
+#[nexus_test]
+async fn test_ip_pool_range_overlapping_ranges_fails(
+    cptestctx: &ControlPlaneTestContext,
```

Comment:
Nice test! When reading the code above, I had a list of test cases I wanted to see, and you covered them all